content/learn/intermediate/outputs-2.md

```diff
@@ -0,0 +1,249 @@
+between them is how they handle special mentions like `@everyone` and `@here`, and whether to return the ID of the
+message after sending it. We will cover these functions in detail in the following sections, and when to best use them.
+
+Let's get started with the simplest of them all, `sendMessage`. Its syntax is the following:
+
+```go
+{{ sendMessage channel_id message_to_be_sent }}
+```
+
+The `channel_id` is the ID of the channel in which the message is to be sent. If you want to send the message in the
+same channel as the one in which the custom command was triggered, simply set `channel_id` to `nil`.
+
+Intuitively, `message_to_be_sent` denotes the output that is to be sent as a message. For now, we will just say it is a
+string with the content you want to send. We will cover sending embeds in a later section on this page.
+
+### Special Mentions
+
+By default, the bot will escape special mentions like `@everyone`, `@here`, and role mentions. If you want to send a
+message with these mentions, you'll need to tell the bot to not escape them. You do this by using the
+`sendMessageNoEscape` function instead of `sendMessage`.
+
+```go
+{{ sendMessageNoEscape channel_id message_to_be_sent }}
+```
+
+### Returning the Message ID
+
+If you want to store the ID of the message you just sent, for example to later edit it, you use the `sendMessageRetID`
+function. Assign the result of this function to a variable, otherwise it will be rendered to the output and become
+unusable for further usage in your custom command.
+
+```go
+{{ $messageID := sendMessageRetID channel_id message_to_be_sent }}
+```
+
+Naturally, we provide a combination of the `sendMessageRetID` function with the `sendMessageNoEscape` function if you
+want to send a message with special mentions and store the ID of the message.
+
+```go
+{{ $messageID := sendMessageNoEscapeRetID channel_id message_to_be_sent }}
+```
+
+It is quite the mouthful to write, but just roll with it.
+
+## Building Embeds
+
+Embeds are a powerful way to display information in a structured and visually appealing way. They can contain a title,
+description, fields, images, and more. We provide a function to build embeds in a way that closely resembles the
+structure as defined by the Discord API.
+
+We will illustrate this with a sparse example. For a full breakdown of all available fields, please refer to our
+[custom embeds documentation](/docs/reference/custom-embeds).
+
+```go
+{{ $embed := cembed
+    "title" "This is a title"
+    "description" "This is a description."
+    "color" 0xff0000
+    "fields" (cslice
+        (sdict "name" "Field 1" "value" "Value 1" "inline" true)
+        (sdict "name" "Field 2" "value" "Value 2" "inline" true)
+    )
+    "author" (sdict
+      "name" .User.Username
+      "icon_url" (.User.AvatarURL "256")
+    )
+}}
+{{ sendMessage nil $embed }}
+```
+
+<div style="float: right; margin-left: 7px; margin-bottom: 10px;">
```

Comment:
Is the raw HTML strictly necessary? I'd prefer to leave it out if possible.

Reply:
The image is rather small—I'd like to keep it to the right of the text. I'm not aware of a markdown / shortcode way.

Reply:
Fair enough, resolving.